libtiledbvcf/src/dataset/tiledbvcfdataset.cc

```diff
@@ -377,11 +397,68 @@ void TileDBVCFDataset::load_field_type_maps(const tiledb::Context& ctx) {
   }
 }
 
+void TileDBVCFDataset::register_samples_v4(const RegistrationParams& params) {
+  if (!open_)
+    throw std::invalid_argument(
+        "Cannot register samples; dataset is not open.");
+
+  assert(metadata_.version == TileDBVCFDataset::V4);
+
+  Config cfg;
+  utils::set_tiledb_config(params.tiledb_config, &cfg);
+  Context ctx(cfg);
+  VFS vfs(ctx, cfg);
+
+  std::vector<SampleAndIndex> samples = SampleUtils::build_samples_uri_list(
+      vfs, params.sample_uris_file, params.sample_uris);
+  if (samples.empty())
+    throw std::invalid_argument(
+        "Cannot register samples; samples list is empty.");
+
+  // Registration proceeds in batches, double-buffering the downloads.
+  std::set<std::string> sample_set(
+      metadata_.all_samples.begin(), metadata_.all_samples.end());
+  std::map<std::string, std::string> sample_headers;
```

Comment:
There is no need for them to be ordered. I don't think we have any maps in anywhere that are needed. Any sorting we do is on vectors.

Reply:
@joe-maley I opened a separate story for this. There is a lot of map usage and in the new v4 system not much of it needs to be ordered. I feel its better to do this in a follow up PR because many basic functions and class attributes needs to be adjusted if we do it across the board, which I think we should.